File: server/vue/src/api/terminal_admin/group.js
import request from '@/utils/request'

export const getGroupTree = () => request.get('/group/tree')
export const getGroupDetail = (id) => request.get(`/group/list/${id}`)
export const addGroup = (data) => request.post('/group/add', data)
export const updateGroup = (id, data) => request.put(`/group/edit/${id}`, data)
export const deleteGroup = (id) => request.delete(`/group/delete/${id}`)
export const getGroupUserTree = () => request.get('/group/user-tree')

// 邀请码
// 邀请码列表
export const getInviteList = () => request.get('/group/invite/list')
// 新增邀请码
export const addInvite = (data) => request.post('/group/invite/add', data)
// 编辑邀请码
export const updateInvite = (id, data) => request.put(`/group/invite/edit/${id}`, data)
// 删除邀请码
export const deleteInvite = (id) => request.delete(`/group/invite/delete/${id}`)

File: server/fastapi/routers/terminal_admin/group.py
from fastapi import APIRouter, Depends, Path

from model.terminal_admin.group_model import GroupCreateUpdate
from service.terminal_admin.group_service import get_group_tree_service, get_group_detail_service, add_group_service, \
    update_group_service, delete_group_service, get_group_with_user_tree_service
from utils.check_perm import check_permission
from utils.log.log_context import log_context_dependency

router = APIRouter()

# 获取分组树结构
@router.get("/tree")
def get_group_tree(_=Depends(log_context_dependency), _p=Depends(check_permission("group:tree"))):
    return get_group_tree_service()

# 获取分组详情
@router.get("/list/{group_id}")
def get_group_detail(group_id: int = Path(..., description="分组ID"), _p=Depends(check_permission("group:detail"))):
    return get_group_detail_service(group_id)

# 新增分组
@router.post("/add")
def add_group(group: GroupCreateUpdate, _=Depends(log_context_dependency),
    _p=Depends(check_permission("group:add"))):
    return add_group_service(group)

# 编辑分组
@router.put("/edit/{group_id}")
def update_group(group_id: int, group: GroupCreateUpdate,_=Depends(log_context_dependency),
    _p=Depends(check_permission("group:edit"))):
    return update_group_service(group_id, group)

# 删除分组
@router.delete("/delete/{group_id}")
def delete_group(group_id: int, _=Depends(log_context_dependency),
    _p=Depends(check_permission("group:delete"))):
    return delete_group_service(group_id)

@router.get("/user-tree")
def get_group_with_user_tree(_=Depends(log_context_dependency)):
    return get_group_with_user_tree_service()
